app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -71,6 +129,270 @@ class UserProfileActivity : AppCompatActivity()
+
+                // The following if-statements are all about adding the edited fields to a map in the view model
+
+                if(binding.userProfileBio.text != binding.userProfileBioEdit.text)
+                {
+                    userProfileViewModel.addEditedField("bio", binding.userProfileBioEdit.text.toString())
+                }
+
+
+                // If at least one field was edited by the user, then send the PUT request to the backend so that the user's changes can be saved
+                if(!userProfileViewModel.isEditedFieldsEmpty())
+                {
+                    userProfileViewModel.updateUserProfile()
+                }
+            }
+        }
+
+        binding.userProfilePasswordButton.setOnClickListener{
+            val builder = AlertDialog.Builder(this)
+            builder.setTitle("Change Password")
+
+            val passwordInput1 = EditText(this)
+            passwordInput1.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+            passwordInput1.hint = "Enter Current Password"
+
+            val passwordInput2 = EditText(this)
+            passwordInput2.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+            passwordInput2.hint = "Enter New Password"
+
+            val passwordInput3 = EditText(this)
+            passwordInput3.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+            passwordInput3.hint = "Confirm New Password"
+
+            val messageText = TextView(this)
+            messageText.setTextColor(Color.RED)
+
+            val layout = LinearLayout(this)
+
+            val params = LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.MATCH_PARENT
+            )
+
+            layout.orientation = LinearLayout.VERTICAL
+            layout.setPadding(50, 75, 50, 75)
+            layout.layoutParams = params
+
+            layout.addView(passwordInput1)
+            layout.addView(passwordInput2)
+            layout.addView(passwordInput3)
+            layout.addView(messageText)
+
+
+            builder.setView(layout)
+
+            builder.setPositiveButton(
+                "Save",
+                DialogInterface.OnClickListener { dialog_ : DialogInterface, _ : Int ->
+
+                })
+
+            builder.setNegativeButton(
+                "Cancel",
+                DialogInterface.OnClickListener { dialog : DialogInterface, _ : Int ->
+                    run {
+                        // When the "Accept button is pressed, simply dismiss the dialog
+                        dialog.dismiss()
+                    }
+                })
+
+            val dialog = builder.create()
+
+            dialog.show()
+
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(View.OnClickListener {
+
+                var closeDialog = false
+
+
+                var validationError : String? = null
```

Comment:
This is redundant since you make the type of the variable nullable and you set its initial value to null. Remove the '= null' part